### ISSUE TITLE
Since 0.9.20, LRR can't pick up archives reliably (only if they are placed in a subfolder with many other archives?)

### ISSUE THREAD

**Issue by soarpasser** [Difegue/LANraragi#1069]
(edited) with further testing this seems to be a completely different issue, marked below.

**Comment by Difegue** [comment]
As mentioned on Discord, I've not been able to repro this.  
![image](https://github.com/user-attachments/assets/46a11576-178f-4829-bd94-6809431ad7db)
  
Might be a W10-only issue? It'd be weird to me though..

**Comment by soarpasser** [comment]
> As mentioned on Discord, I've not been able to repro this. ![image](https://private-user-images.githubusercontent.com/8237712/363444897-46a11576-178f-4829-bd94-6809431ad7db.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjUxNDk0NzIsIm5iZiI6MTcyNTE0OTE3MiwicGF0aCI6Ii84MjM3NzEyLzM2MzQ0NDg5Ny00NmExMTU3Ni0xNzhmLTQ4MjktYmQ5NC02ODA5NDMxYWQ3ZGIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDkwMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA5MDFUMDAwNjEyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OGQ0MGY0ZjRhMTk4YWY5YWZiM2EzM2I1NjQ3Njc2NTJhNTQ5MzU4MGVlNGY3OTUwOWVjNGQ5OTQ0MzNkYTIwZCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.P0wYTbcSiwVf085CNdjXE42rClFwxqahfqTP4ZIusSU)
> 
> Might be a W10-only issue? It'd be weird to me though..

Will need testing on a windows 10 environment. I can't get my windows 10 VM to work with the program.

**Comment by soarpasser** [comment]
After further testing, I can confirm that this issue appears on Windows 10, WSL2 installations of versions 0.9.20 and up.
The issue does not occue on Windows 10, WSL1 installations.

**Comment by soarpasser** [comment]
![image](https://github.com/user-attachments/assets/adc665ba-e669-4459-8693-374fcdde59a8)
![image](https://github.com/user-attachments/assets/1986e51d-f179-4341-a249-328b3d7a44ba)


Curiously, after renaming the content folder a couple of times and triggering dozens of rescanning, this happened.
I managed to get the program to pick up on the archive that originally couldn't get picked up at all, but now it isn't showing up in search for some reason.

**Comment by soarpasser** [comment]
Did even more testing...
It looks like this problem only happens when you move the archive to a specific subfolder under the content folder.
More specifically, I think it has to do with the number of items that are currently in that subfolder.

Here's how I organize my content folder:

1. Root folder
   2. subfolder1 (4000+files)
   3. subfolder2 (800 files)
   4. subfolder3 (600 files)
   5. ......
   10. subfolder10 (20 files)

I find that when, and ONLY when I move an archive in subfolder1 with a large amount of files, does this error have the potential to occur. Otherwise, when I move it anywhere else, no matter it be a subfolder or the root content directory, the error does not occur.

Thus I suspect that the amount of files in the content folder might cause this issue to happen. 
This also means that this might not just be limited to windows 10 as originally suspected, but instead to content folders that fulfill the above criteria.

**Comment by siliconfeces** [comment]
This is conjecture, but I do have some suspicions.

When LRR does it initial scan, it first finds all files before doing any processing. Thus, if there's a lot of files it may take a while before anything happens. ALSO, if any archive causes LRR to crash, you won't get any archives.

If you look in the logs (Shinobu being particularly interesting), does anything stand out?

**Comment by soarpasser** [comment]
> This is conjecture, but I do have some suspicions.
> 
> When LRR does it initial scan, it first finds all files before doing any processing. Thus, if there's a lot of files it may take a while before anything happens. ALSO, if any archive causes LRR to crash, you won't get any archives.
> 
> If you look in the logs (Shinobu being particularly interesting), does anything stand out?

No, even with debug mode enabled it doesn't show any relevant logs, it's like nothing happens at all.
I've since went to legacy builds running WSL1 and the issue doesn't exist there.

**Comment by hucurve** [comment]
I also encountered a similar problem, so that all the files in the directory rather than sub-file will solve this problem？thats really annoyning

**Comment by IAmCaligvla** [comment]
Can confirm, LRR isn't picking up new archives. I have to manually add all new files to the database, one by one.

**Comment by Guerra24** [comment]
@soarpasser @hucurve @IAmCaligvla Can anyone confirm if this is still an issue on 0.9.60?

**Comment by IAmCaligvla** [comment]
Can't really speak for the others, but on the current nightlies for Windows I haven't had any problems with file detection at all.
